x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -49,12 +49,12 @@ protected Object unwrapCustomValue(Object values) {
 
         if (dataType == DATETIME) {
             if (values instanceof String) {
-                return parseDateString(values);
-            }
-        }
-        if (dataType == DATETIME_NANOS) {
-            if (values instanceof String) {
-                return DateUtils.asDateTimeWithNanos(values.toString(), zoneId());
+                try {
+                    return DateUtils.asDateTimeWithNanos(values.toString(), zoneId());
```

Comment:
I'd wait for the fields API from @astefan to land before going for this one.
Hopefully that eliminates the internal handling of data format and thus this try/catch as well.

Reply:
I guess fields API will make this uniform, but just in case, an alternative might be to check the presence of the `T` or `.` in the string and based on that call the right parser.

Reply:
I was curious to see the perf difference of try/catch and trying to check the string:
```
Benchmark                              Mode  Cnt        Score      Error  Units
BenchmarkDateParse.testWithContains   thrpt   25  1631774,366 ± 6016,080  ops/s
BenchmarkDateParse.testWithException  thrpt   25   629767,727 ± 3843,439  ops/s
```
and it's quite big!

Reply:
I will use the `contains("-")` since it's always there for Datetime stings and it's at the beginning of the string.

Reply:
> and it's quite big!

I wonder if indexing the string wouldn't actually make a further difference: contains takes each character at a time, but the datetime format is fixed, so `-` should always be at `.charAt(4)`. Either way, I think it's fine.

Reply:
hm, yeah, I rushed here, need to revisit:

- a timestamp can start with `-` (negative millis).
- `.charAt(4)` is also not safe, as there are possible dates like: `+14953-02-....` or `-134-05-....`